src/components/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,65 @@
+import classNames from 'classnames/bind';
+import { Fragment } from 'react';
+
+import Modal from '../Modal/Modal';
+import Button from '../Buttons/Button/Button';
+import { DialogIcon } from './DialogIcon';
+
+import styles from './Dialog.module.scss';
+
+const cn = classNames.bind(styles);
+
+interface BaseDialogProps {
+  message: string;
+  isOpen: boolean;
+  iconType?: 'warn' | 'accept';
+}
+
+interface AlertDialogProps extends BaseDialogProps {
+  type: 'alert';
+  buttonText: string;
+  onClick: () => void;
+}
+
+interface ConfirmDialogProps extends BaseDialogProps {
+  type: 'confirm';
+  buttonText: Record<'left' | 'right', string>;
+  onClick: Record<'left' | 'right', () => void>;
+}
+
+type DialogProps = AlertDialogProps | ConfirmDialogProps;
+
+export default function Dialog({ type, isOpen, onClick, buttonText, iconType, message }: DialogProps) {
+  return (
+    <Modal isOpen={isOpen} onClose={() => {}}>
```

Comment:
```suggestion
    <Modal isOpen={isOpen} onClose={() => {}}>
```
onClose 보기가 좀 그런데 Modal에서 props을 옵셔널로 수정하는게 어떠세요